src/app/page.tsx

```diff
@@ -1,21 +1,76 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import InfiniteScroll from "react-infinite-scroll-component";
 import { getTopStories } from "@/services/hn";
 import Link from "next/link";
 
-export default async function Home() {
-  const topstories = await getTopStories();
+const ITEMS_PER_PAGE = 10; // Number of items displayed per page in paginated views from API
+
+const Home = () => {
+  const [stories, setStories] = useState<any[]>([]);
+  const [hasMore, setHasMore] = useState(true);
+  const [page, setPage] = useState(0);
+
+  const fetchMoreStories = async () => {
```

Comment:
You're actually fetching the top 10 stories over and over again and applying them to the list of articles so it's just 10 unique articles in the whole list. :)

What you also need to do is to alter the hn service and apply logic to the pagination there as well.

![image](https://github.com/user-attachments/assets/5c4fe20e-6fc5-4717-8e5e-1dbe149d56e4)